src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -235,8 +257,19 @@ public boolean equals(Object other) {
 
         // state check
         EditCommand e = (EditCommand) other;
-        return index.equals(e.index)
-                && editPersonDescriptor.equals(e.editPersonDescriptor);
+
+        // both have an editStudentDescriptor
+        if (editPersonDescriptor == null && e.editStudentDescriptor != null) {
```

Comment:
is there a case where both are not null (e.g. editing a student's original person fields)?

Reply:
nope, there are two constructor methods: one is to instantiate the EditCommand with an EditPersonDescriptor and the other instantiates with an EditStudentDescriptor. there is no case where both are null. also since both fields are final, there is no way to modify the EditCommand once instantiated.

Reply:
there is also no case where both are not null, because the `editCommandParser` will only call one of the `EditCommand` constructors. With regards to your example, if you edit a students original person fields, the `EditCommand` will have an `editPersonDescriptor`, which will only be converted upon calling `execute()`

Reply:
ah I see!